underfs/swift/src/main/java/alluxio/underfs/swift/KeystoneV3AccessProvider.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.swift;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.javaswift.joss.client.factory.AccountConfig;
+import org.javaswift.joss.client.factory.AuthenticationMethod.AccessProvider;
+import org.javaswift.joss.model.Access;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Custom {@link AccessProvider} for Keystone V3 authentication.
+ */
+public class KeystoneV3AccessProvider implements AccessProvider {
+  private static final Logger LOG = LoggerFactory.getLogger(KeystoneV3AccessProvider.class);
+
+  private static final String AUTH_METHOD = "password";
+  private static final int RESPONSE_OK = 201;
+
+  private AccountConfig mAccountConfig;
+
+  /**
+   * Create a new instance of {@link KeystoneV3AccessProvider}.
+   *
+   * @param accountConfig account credentials
+   */
+  public KeystoneV3AccessProvider(AccountConfig accountConfig) {
+    mAccountConfig = accountConfig;
+  }
+
+  @Override
+  public Access authenticate() {
+
+    try {
+      String requestBody;
+      try {
+        // Construct request body
+        KeystoneV3Request request =
+            new KeystoneV3Request(new Auth(
+                new Identity(Arrays.asList(AUTH_METHOD),
+                    new Password(
+                        new User(mAccountConfig.getUsername(), mAccountConfig.getPassword()))),
+                new Scope(new Project(mAccountConfig.getTenantName()))));
+        requestBody = new ObjectMapper().writeValueAsString(request);
+      } catch (JsonProcessingException e) {
+        LOG.error("Error processing JSON request: {}", e.getMessage());
+        return null;
+      }
+
+      HttpURLConnection connection = null;
+      BufferedReader bufReader = null;
+      try {
+        // Send request
+        connection = (HttpURLConnection) new URL(mAccountConfig.getAuthUrl()).openConnection();
```

Comment:
correct, I think we should be consistent in our code base and use Apache HttpClient (see https://www.mkyong.com/java/how-to-send-http-request-getpost-in-java/ for reference)